Add a new UI with webworker REPL support

katharosada opened the pull request:
I've kept it as minimal as I can, but for now it works ...mostly! Just sending this PR for feedback.

Notes:
* Copies the `python.js`, `python.wasm` and `python.data` files into the folder - happy to find a better way to do this (symlink?)
* Supports REPL only, I've not made it use the `<textarea>` yet so trying to click the 'Run' button will just error out.
* No matter what I tried, I cant get stdin to be auto-detected as a tty (apparently `isatty` always returns false in emscripten) but using the args `-i -` forces interactive mode.
* Not very performant on stdout/stderr since it's individually sending each character via `postMessage` lol. (It'd be better to buffer it and use postMessage only on flush).
* Very limited terminal control code handling, only supports typing visible characters and backspace (won't deal well with line wrapping)
* Doesn't handle copy/paste... at all.
* Attempts to encode 1 byte at a time into ASCII - understandably this completely breaks any non-ascii utf-8

tiran commented on browser-ui/index.html:
```diff
@@ -0,0 +1,22 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>wasm-python terminal</title>
+    <link rel="stylesheet" href="https://unpkg.com/xterm@4.16.0/css/xterm.css" crossorigin/>
+    <link rel="stylesheet" href="/styles.css"/>
```
Make import relative so we can deploy the shell in a sub directory on Ethan's GH page.
```suggestion
    <link rel="stylesheet" href="./styles.css"/>
```

tiran commented on browser-ui/index.html:
```diff
@@ -0,0 +1,22 @@
+    </div>
+    <div id="terminal"></div>
+</body>
+</html>
```
Missing newline at EOF
```suggestion
</html>

```

tiran commented on browser-ui/server.py:
```diff
@@ -0,0 +1,21 @@
+        self.send_header("Cross-Origin-Embedder-Policy", "require-corp")
+
+
+with socketserver.TCPServer(("", PORT), MyHTTPRequestHandler) as httpd:
```
``http.server`` uses ``ThreadingTCPServer``

You could also use ``http.server.test(HandlerClass=MyHTTPReqyestHandler, protocol="HTTP/1.1")``

katharosada replied:
Happy to make the change, but um, what's the difference?

katharosada replied:
Actually, never mind I get it now. :)

tiran reviewed (commented):
Fantastic! Thanks a lot.

I noticed that all files are missing a newline at end of file.

ethanhs reviewed (approved):
This looks fantastic! I don't have anything to add other than +1 to Christian's review comments. Very much looking forward to landing this :)

kumaraditya303 commented on browser-ui/server.py:
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python
+
+import socketserver
+from http import server # Python 3
+
+PORT = 8000
```
```suggestion
from http import server # Python 3
import os

PORT = int(os.getenv("PORT", 8000))
```

tiran replied:
If we decide to make the port configurable, then we should use argparse instead. For now hard-coded port is good enough.

kumaraditya303 commented on browser-ui/server.py:
```diff
@@ -0,0 +1,21 @@
+    def end_headers(self):
+        self.send_my_headers()
+
+        server.SimpleHTTPRequestHandler.end_headers(self)
```
```suggestion
        super().end_headers()
```

katharosada commented:
Fixed up all the things! Even added `argparse` to specify the port for the http server.
Also, I removed the textarea and `Run` button since I didn't implement them yet. I can put that in a separate PR.

ethanhs commented:
Thank you so much!
